libsolidity/formal/SMTChecker.h

```diff
@@ -163,9 +163,44 @@ class SMTChecker: private ASTConstVisitor
 	VariableIndices visitBranch(ASTNode const* _statement, smt::Expression const* _condition = nullptr);
 	VariableIndices visitBranch(ASTNode const* _statement, smt::Expression _condition);
 
+	using CallStackEntry = std::pair<CallableDeclaration const*, ASTNode const*>;
+
+	/// Verification targets.
+	//@{
+	struct VerificationTarget
+	{
+		enum class Type { ConstantCondition, Underflow, Overflow, UnderOverflow, DivByZero, Balance, Assert } type;
+		smt::Expression value;
+		smt::Expression constraints;
```

Comment:
Are these the path constraints?

Reply:
Not only, it's all the constraints at the moment of target creation. This makes the check independent from the runtime encoding context